mono/mini/exceptions-arm64.c

```diff
@@ -620,3 +626,598 @@ mono_arch_do_ip_adjustment (MonoContext *ctx)
 	pc = (gpointer)((guint64)MINI_FTNPTR_TO_ADDR (pc) - 1);
 	ctx->pc = (host_mgreg_t)MINI_ADDR_TO_FTNPTR (pc);
 }
+
+
+#ifdef MONO_ARCH_HAVE_UNWIND_TABLE
+
+static void 
+mono_arch_unwind_add_assert (guint32 unwind_code_size, guint32 max_offset, guint32 unwind_codes_buffer_size, guint32 offset) {
+	g_assert(offset <= max_offset);
+	if (unwind_codes_buffer_size + unwind_code_size > MONO_ARM64_MAX_UNWIND_CODE_SIZE)
+		g_error ("Larger allocation needed for the unwind information.");
+}
+
+static void
+mono_arch_unwind_add_nop (guint8* unwind_codes, guint32* unwind_code_size) {
+	mono_arch_unwind_add_assert(1, 0, *unwind_code_size, 0);
+	unwind_codes[(*unwind_code_size)++] = 0b11100011;
+}
+
+static void
+mono_arch_unwind_add_nops (guint8* unwind_codes, guint32* unwind_code_size, int nop_count) {
+	while (nop_count--)
+		mono_arch_unwind_add_nop (unwind_codes, unwind_code_size);
+}
+
+
+static void
+mono_arch_unwind_add_set_fp (guint8* unwind_codes, guint32* unwind_code_size) {
+	mono_arch_unwind_add_assert (1, 0, *unwind_code_size, 0);
+	unwind_codes[(*unwind_code_size)++] = 0b11100001;
```

Comment:
Nit, a lot of magic numbers in this diff. Not sure how much is to be consistent with other places in Mono, or whether we want to do more.

Reply:
This magic number are from the windows unwind codes - https://learn.microsoft.com/en-us/cpp/build/arm64-exception-handling?view=msvc-170#unwind-codes.  They'll exist only here, not sure that's much benefit in making them defines.

Reply:
I've added comments from the documentation on the format and meaning of these codes in https://github.com/Unity-Technologies/mono/pull/1960/commits/9b8e291ee6dde9c51871bf05de6ad195e1d48f01.  That should help.